pf/tests/testdata/basicprogram/Pulumi.yaml

```diff
@@ -94,6 +94,22 @@ resources:
       tuplesOptionals:
         - { t0: false, t1: "no" }
         - { t0: true, t1: "yes"}
+  setElementAttrRes:
+    type: testbridge:index:Testres
+    properties:
+      statedir: state
+      requiredInputString: "x"
+      setOptionals:
+        - "hi"
+        - "bye"
+  setElementRevAttrRes:
+    type: testbridge:index:Testres
+    properties:
+      statedir: state
+      requiredInputString: "x"
+      setOptionals:
+        - "bye"
```

Comment:
Hm it preserves order?

Reply:
It seems like it. Sets are implemented as `[]tftypes.Value`, so we seem to get array-like properties. This would all be very different if sets were implemented as `map[ValueHash]Value`.

Reply:
Huh that's good to know. Interesting. I think in TF making data as a set mostly affects how they go through diffing operations, sounds like the repr isn't helpful to suss out how this works.